modules/parent-join/src/main/java/org/elasticsearch/join/aggregations/ChildrenAggregationBuilder.java

```diff
@@ -112,14 +112,20 @@ protected void innerWriteTo(StreamOutput out) throws IOException {
 
     private void joinFieldResolveConfig(SearchContext context, ValuesSourceConfig<WithOrdinals> config) {
         ParentJoinFieldMapper parentJoinFieldMapper = ParentJoinFieldMapper.getMapper(context.mapperService());
-        ParentIdFieldMapper parentIdFieldMapper = parentJoinFieldMapper.getParentIdFieldMapper(childType, false);
-        if (parentIdFieldMapper != null) {
-            parentFilter = parentIdFieldMapper.getParentFilter();
-            childFilter = parentIdFieldMapper.getChildFilter(childType);
-            MappedFieldType fieldType = parentIdFieldMapper.fieldType();
-            final SortedSetDVOrdinalsIndexFieldData fieldData = context.getForField(fieldType);
-            config.fieldContext(new FieldContext(fieldType.name(), fieldData, fieldType));
-        } else {
+        if(parentJoinFieldMapper != null){
```

Comment:
Hmm, let's see if we can un-nest the `if` conditional?  Usually cleaner to read if they don't nest in my opinion.  How about something like:

```java
if (parentJoinFieldMapper == null) {
    config.unmapped(true);
    return;
}

ParentIdFieldMapper parentIdFieldMapper = parentJoinFieldMapper.getParentIdFieldMapper(childType, false);
if (parentIdFieldMapper == null) {
    config.unmapped(true);
    return;
}
```

And then after that point we know both fields are not null, so we can set the config outside of a conditional.  Should flatten the whole thing a bit.

Reply:
Alright, makes sense